### ISSUE TITLE
Importing data in COCO format

### ISSUE THREAD

**Issue by amitguptagwl** [NaturalIntelligence/imglab#82]
> This feature is suitable for first-timers. But can be picked by any contributor.

Currently, imglab supports exporting labeled data in COCO format. We want to let users importing data in COCO format.

target file : [coco.js](https://github.com/NaturalIntelligence/imglab/blob/master/dataformaters/coco.js)

Please check how the export feature is implemented in [coco.js](https://github.com/NaturalIntelligence/imglab/blob/master/dataformaters/coco.js)

> [Watch](https://github.com/NaturalIntelligence/imglab/watchers) for changes, or [Bookmark](https://github.com/NaturalIntelligence/imglab/stargazers) for easy discovery.
> [Fund](https://www.patreon.com/bePatron?u=9531404) this project for new features and maintenance.
> [Showcase] your project with us by raising an issue

**Comment by amitguptagwl** [comment]
Yes sure. Let me know if you need any guidance

**Comment by amitguptagwl** [comment]
@jbovee any update?

**Comment by jbovee** [comment]
I'm progressing pretty well. I've just about got the `fromCOCO` function done, and made the change to `openfile.js` in order to be able to test it. I noticed that the COCO format doesn't store the type of polygon (rect, circle, polygon) for each shape. Will there be an issue with importing if the objects in the resulting `shapes` array have a `type` value of `undefined`?

Also, I noticed an error in the toCOCO function, where `shape_i.bbox` was used instead of `shape.bbox` causing the `bbox` field to always be `undefined`. Should I put the fix to that in a separate PR, or in the same one I'll eventually have for filling in the `fromCOCO` function?

**Comment by amitguptagwl** [comment]
1. As the shape type is not being saved. Use bbox and create rectangle shape from that.
2. Please raise a PR for the fix first then PR for COCO import.

**Comment by jbovee** [comment]
I've got the function completed, and my tests of it seem to work fine. I'll raise the two PRs later this evening for you to check.

**Comment by jbovee** [comment]
Actually, I have a question I almost forgot about. Is the polygon tool functional? I can't figure out how to get it to stop creating points and finish the polygon. And because of that I haven't been able to thoroughly test it.

**Comment by amitguptagwl** [comment]
Yes, it is working. You'll have to press enter to finish it.

**Comment by jbovee** [comment]
I've raised the two PR's; one for fixing the `bbox` error in `toCOCO`, and the other for supporting importing COCO data.

After taking a closer look at the way each polygon type was represented in the COCO formatted data, I realized that although the type of polygon isn't explicitly stored, it is possible to determine the type based on factors such as the length of the points array inside `segmentation`, the area of the polygon, and how the values in the points array compare to the values in `bbox`.

So as you'll see in the PR, if the points array has a length of two, both of those values correspond to `cx` and `cy` in `bbox`, and the area is not 0, then the polygon is a circle. And unless the points array has a length of eight, and all of the points exactly match the rectangle layout of `[x, y, x+w, y, x+w, y+h, x, y+h]`, then it is just a normal polygon.
